src/main/java/org/broadinstitute/hellbender/tools/spark/sv/utils/SVReferenceUtils.java

```diff
@@ -0,0 +1,116 @@
+                .map(SAMSequenceRecord::getSequenceName).collect(Collectors.toList());
+    }
+
+    /**
+     * Create an RDD from the reference sequences.
+     * The reference sequences are transformed into a single, large collection of byte arrays. The collection is then
+     * parallelized into an RDD.
+     * Each contig that exceeds a size given by REF_RECORD_LEN is broken into a series of REF_RECORD_LEN chunks with a
+     * K-1 base overlap between successive chunks. (I.e., for K=63, the last 62 bases in chunk n match the first 62
+     * bases in chunk n+1) so that we don't miss any kmers due to the chunking -- we can just kmerize each record
+     * independently.
+     */
+    public static JavaRDD<byte[]> getRefRDD(final JavaSparkContext ctx,
+                                            final int kSize,
+                                            final ReferenceMultiSource ref,
+                                            final SAMSequenceDictionary readsDict,
+                                            final int ref_record_len,
+                                            final int ref_records_per_partition) {
+        final SAMSequenceDictionary dict = ref.getReferenceSequenceDictionary(readsDict);
+        if ( dict == null ) throw new GATKException("No reference dictionary available");
+
+        final int effectiveRecLen = ref_record_len - kSize + 1;
+        final List<byte[]> sequenceChunks = new ArrayList<>();
+        for ( final SAMSequenceRecord rec : dict.getSequences() ) {
+            final String seqName = rec.getSequenceName();
+            final int seqLen = rec.getSequenceLength();
+            final SimpleInterval interval = new SimpleInterval(seqName, 1, seqLen);
+            try {
+                final byte[] bases = ref.getReferenceBases(null, interval).getBases();
+                for ( int start = 0; start < seqLen; start += effectiveRecLen ) {
+                    sequenceChunks.add(Arrays.copyOfRange(bases, start, Math.min(start+ref_record_len, seqLen)));
+                }
+            }
+            catch ( final IOException ioe ) {
+                throw new GATKException("Can't get reference sequence bases for " + interval, ioe);
+            }
+        }
+
+        return ctx.parallelize(sequenceChunks, sequenceChunks.size()/ref_records_per_partition+1);
+    }
+
+    /**
+     * Do a map/reduce on an RDD of genomic sequences:
+     * Kmerize, mapping to a pair <kmer,1>, reduce by summing values by key, filter out <kmer,N> where
+     * N <= MAX_KMER_FREQ, and collect the high frequency kmers back in the driver.
+     */
+    @VisibleForTesting
+    public static List<SVKmer> processRefRDD(final int kSize,
+                                             final int maxDUSTScore,
+                                             final int maxKmerFreq,
+                                             final JavaRDD<byte[]> refRDD) {
+        final int nPartitions = refRDD.getNumPartitions();
```

Comment:
Check arguments are not null or otherwise invalid.

Reply:
I think that getRefRDD and processRefRDD are doing a very specific kind of work on the reference genome and so I don't think they would qualify as utilities ... they seem to be only used by FindBadGenomeKmers, so I would suggest that these methods are moved to that class.

Reply:
checked.

Reply:
`getRefRDD` was in `SVUtils` before so I moved it here thinking it is providing a special type of utility related to reference. So I'll leave it here.
Moved processRefRDD back to `FindBadGenomeKmers`